### PR TITLE
significant progress on ratings dashboard

### PR DESCRIPTION
**Pull request by kevinleaves**
- displaying bars based on % star rating
- user can now additively filter reviews based on star rating on bar click.
- total # of reviews are displayed, (still need to dynamically update this number when user filters by review)
- display count of each star rating

### CODE REVIEW

File: client/src/components/RatingsReviews.jsx [hunk 1]
```diff
@@ -26,15 +34,33 @@ export default function RatingsReviews({ productID, reviewMetadata, product }) {
     setSort(event.target.value);
   };
 
+  const toggleSelectedRating = (starAmt) => {
+    setSelectedRatings({
+      ...selectedRatings,
+      [starAmt]: !selectedRatings[starAmt],
```
Comment (quanjingchen) [inline client/src/components/RatingsReviews.jsx:40]: i like how you toggle the star rating!

File: client/src/components/Sub_RatingsReviews/Dashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export default function Dashboard({
+  reviews, reviewMetadata, toggleSelectedRating, selectedRatings,
+}) {
+  const totalAmtOfReviews = Object.values(reviewMetadata.ratings)
+    .reduce((accumulator, currrent) => Number(currrent) + accumulator, 0);
+
+  const totalStarRating = Object.entries(reviewMetadata.ratings)
+    .map((entry) => Number(entry[0]) * Number(entry[1]))
+    .reduce((accumulator, current) => accumulator + current, 0);
```
Comment (quanjingchen) [inline client/src/components/Sub_RatingsReviews/Dashboard.jsx:13]: I like the way you use underscore function! neat!

File: client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+export default function RatingBreakdown({
+  reviewMetadata, totalAmtOfReviews, toggleSelectedRating, selectedRatings,
+}) {
+  const progress = Object.entries(reviewMetadata.ratings)
```
Comment (quanjingchen) [inline client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx:6]: using object.entries is a super smart way to get key value pair!

File: client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        const percentage = Math.round(entry[1] * 100);
+        return (
+          <div
+            key={entry[0]}
+            className="rating-container"
+            onClick={() => toggleSelectedRating(entry[0])}
+          >
+            <div className="rating-bar-label">{entry[0]} stars:</div>
+            <progress
+              className="rating-bar"
```
Comment (quanjingchen) [inline client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx:25]: i'm a little confused here. progress is not a react component, how can you do <progress /> and assign a classname to it?

Comment (kevinleaves) [reply client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx:25]: hi Quanjing, I was referring to this doc.

https://beta.reactjs.org/reference/react-dom/components/progress

It looks like react does consider it a component, but I _did_ have some issues yesterday styling it, so I think there's some weird behavior that I still have to figure out with this element type.

Comment (quanjingchen) [reply client/src/components/Sub_RatingsReviews/RatingBreakdown.jsx:25]: interesting...

**Pull request merged**